src/xcode/ENA/ENA/Source/Services/Risk/Provider/Helper/CachedAppConfiguration.swift

```diff
@@ -66,54 +66,53 @@ final class CachedAppConfiguration {
 	private var promises = [(Result<CachedAppConfiguration.AppConfigResponse, Never>) -> Void]()
 	private var requestIsRunning: Bool { !promises.isEmpty }
 
+	private static let queue = DispatchQueue(label: "fetchAppConfiguration queue", qos: .userInitiated, attributes: .concurrent)
```

Comment:
Please correct me, but in my understanding a .concurrent queue in combination with .barrier for every task in this queue, makes it a serial queue. Why not using a serial queue without .barrier?

Reply:
@ArturFriesen Check my comment above. Great minds think alike ;-)

Reply:
No it is not the same. 
Using the .barrier flag, lets the the running "jobs" in the Queue finish. Then the Queue is blocked until the "job" with the` .barrier` flag is completed. So while the job with the .barrier flag is not running the queue is concurrent.

Reply:
But in this implementation it makes no difference, right?

Reply:
It makes a difference, since we don't bottle neck the queue while we are adding requests to the cachedappconfig. Those can run from n threads at once.

Reply:
But if every operation on the queue is a barrier task then you could just make the queue a serial one. As far as I saw that was the only kind of operation executed on that queue.

Reply:
Oh I didn't read the whole changes, my bad, we had a call about the change and I thought only the resolving of the promises should be with the barrier flag 🤔